### PR TITLE
Created new export methods and updated notebooks

### LINKED ISSUE #1226: Update vis module to conversions
- Add all properties in the graph to the networkx graph
- Return the networkX graph instead of directly visualising it 
- Remove any unneeded arguments to the `to_networkx` function
- Change the name of vis to `export` or something similar

### PR DESCRIPTION
**Pull request by rachchan**
### What changes were proposed in this pull request?
- Changed Vis file to "export" and added to_networkx, to_edge_list_df, to_node_list_df methods. Minor fix to pyvis method as new property API.
- updated notebooks accordingly

### Why are the changes needed?
- so people can export raphtory graphs into networkx graphs and panda dataframes

### Does this PR introduce any user-facing change? If yes is this documented?
Yes all documented

### How was this patch tested?
Full suite of tests added

### Issues
Fixes #1226 

### Are there any further changes required?
We need to port this over to rust at some point, but is fine here for now.

### CODE REVIEW

File: python/python/raphtory/export.py [hunk 1]
```diff
@@ -117,43 +118,90 @@ def to_pyvis(
...
+
+        vertex_tuple = (name, properties)
+        vertex_tuples.append(vertex_tuple)
+
+    networkXGraph.add_nodes_from(vertex_tuples)
 
-    networkXGraph.add_edges_from(edges)
-    pos = nx.spring_layout(networkXGraph, k=k, iterations=iterations)
+    edge_tuples = []
+    for e in graph.edges():
```
[the remark this answers is not in the document]
Comment (miratepuffin) [reply python/python/raphtory/export.py:154]: `explode_layers`

File: python/tests/notebook.ipynb [hunk 2]
```diff
@@ -2,7 +2,7 @@
  "cells": [
   {
    "cell_type": "code",
-   "execution_count": 9,
+   "execution_count": 1,
```
Comment (miratepuffin) [inline python/tests/notebook.ipynb:5]: Can these be converted into actual tests.

**Review by miratepuffin — DISMISSED** [review]

File: python/python/raphtory/export.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
+"""
+Generate a visualisation using matplotlib or pyvis from Raphtory graphs.
+"""
+from pyvis.network import Network
+import networkx as nx
+import pandas as pd
+
+r"""Draw a graph with Pyvis.
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:8]: This docstring is in the wrong place, it should be after the def

File: python/python/raphtory/export.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+    edge_weight=None,
+    edge_label=None,
+    colour_nodes_by_type=False,
+    type_property="type",
+    notebook=True,
+    **kwargs,
+):
+    """
+    Returns a dynamic visualisation in static HTML format from a Raphtory graph.
+    """
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:68]: replace with the docstring above

File: python/python/raphtory/export.py [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    The ones with the label outside of it are: image, circularImage, diamond, dot, star, triangle, triangleDown, square and icon.
+    By default ``"dot"`` is set.
+:param str node_image: An optional string defining the url of a custom node image. By default an image of a circle is set.
+:param str edge_weight: An optional string defining the name of the property where edge weight is set on your Raphtory graph. By default ``1`` is set.
+:param str edge_label: An optional string defining the name of the property where edge label is set on your Raphtory graph. By default, an empty string as the label is set.
+:param bool notebook: A boolean that is set to True if using jupyter notebook. By default this is set to True.
+
+
+:returns: A pyvis visualisation in static HTML format that is interactive with toggles menu.
+:rtype: IFrame(name, width=self.width, height=self.height)
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:34]: this is wrong, it returns a Pyvis graph not an IFrame (it no longer calls show internally)

File: python/python/raphtory/export.py [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+            value=weight,
+            color=edge_color,
+            title=label,
+            arrowStrikethrough=False,
+        )
+
+    return visGraph
+
+
+r"""Returns a graph with NetworkX.
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:116]: docstring is. in the wrong place

File: python/python/raphtory/export.py [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+            else:
+                properties.update({"update_history": e.history()})
+        edge_tuples.append((src, dst, properties))
+
+    networkXGraph.add_edges_from(edge_tuples)
+
+    return networkXGraph
+
+
+r"""Returns an edge list pandas dataframe fro the given graph.
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:190]: also in the wrong place

Comment (ljeub-pometry) [inline python/python/raphtory/export.py:190]: typo fro -> for

File: python/python/raphtory/export.py [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+                tuple.append(e.time())
+            else:
+                tuple.append(e.history())
+
+        edge_tuples.append(tuple)
+
+    return pd.DataFrame(edge_tuples, columns=columns)
+
+
+r"""Returns an vertex list pandas dataframe fro the given graph.
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:250]: also wrong place

File: python/python/raphtory/graphqlserver.py [hunk 9]
```diff
@@ -41,38 +42,44 @@ def wait_for_online(self):
...
+
+
+async def __from_map_and_directory(graphs, graph_dir, port):
+    await internal_graphql.from_map_and_directory(graphs, graph_dir, port)
+
 
-async def __from_map_and_directory(graphs,graph_dir,port):
-    await internal_graphql.from_map_and_directory(graphs,graph_dir,port)
+async def __from_directory(graph_dir, port):
+    await internal_graphql.from_directory(graph_dir, port)
```
Comment (ljeub-pometry) [inline python/python/raphtory/graphqlserver.py:59]: did we really want `__` here? They are special in python and cause weird name mangling. If you just want to denote this as a private method you should use `_`

File: python/tests/test_graph_conversions.py [hunk 10]
```diff
@@ -0,0 +1,1011 @@
...
+        vertex_props=["OS_version", "primary_function", "uptime_days"],
+        vertex_const_props=["server_name", "hardware_type"],
+        vertex_shared_const_props={"datasource": "data/network_traffic_edges.csv"},
+    )
+
+
+# def test_edge_list_df():
+
+# def test_vertex_list_df():
+
```
Comment (ljeub-pometry) [inline python/tests/test_graph_conversions.py:39]: should actually add these tests

**Review by ljeub-pometry — CHANGES_REQUESTED** [review]
All the new docstrings are in the wrong place and missing tests for the dataframe functions. Otherwise looks good

File: python/tests/test_graph_conversions.py (hunk 10 above)
Comment (miratepuffin) [reply python/tests/test_graph_conversions.py:39]: These are being testing fully, I just forgot to remove the comments - will do now

Comment (ljeub-pometry) [reply python/tests/test_graph_conversions.py:39]: didn't find them scrolling through, probably just missed them in the large diff

File: python/python/raphtory/export.py [hunk 11]
```diff
@@ -187,34 +173,27 @@ def to_networkx(
...
 def to_edge_df(
     graph,
     explode_edges=False,
     include_edge_properties=True,
     include_update_history=True,
     include_property_histories=True,
 ):
-    """
-    Returns a edge list dataframe from a Raphtory graph.
+    r"""Returns an edge list pandas dataframe fro the given graph.
```
Comment (ljeub-pometry) [inline python/python/raphtory/export.py:183]: typo still there

**Review by ljeub-pometry — APPROVED** [review]
Note, typo is still there

**Pull request merged**